test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_simultaneous.cpp

```diff
@@ -429,26 +423,211 @@ struct OutOfOrderTest : public BasicMutableCommandBufferTest
     clProgramWrapper program_fill;
 
     const size_t test_global_work_size = 3 * sizeof(cl_int);
-    cl_mutable_command_khr command = nullptr;
+    const cl_int pattern_pri = 42;
 
     const cl_int overwritten_pattern = 0xACDC;
+    cl_mutable_command_khr command;
+};
+
+struct CrossQueueSimultaneousMutableDispatchTest
+    : public BasicMutableCommandBufferTest
+{
+    CrossQueueSimultaneousMutableDispatchTest(cl_device_id device,
+                                              cl_context context,
+                                              cl_command_queue queue)
+        : BasicMutableCommandBufferTest(device, context, queue),
+          queue_sec(nullptr), command(nullptr)
+    {
+        simultaneous_use_requested = true;
+    }
+
+    cl_int SetUpKernel() override
+    {
+        const char* kernel_str =
+            R"(
+          __kernel void fill(int pattern, __global int* out)
+          {
+              size_t id = get_global_id(0);
+              out[id] = pattern;
+          })";
+
+        cl_int error = create_single_kernel_helper_create_program(
+            context, &program, 1, &kernel_str);
+        test_error(error, "Failed to create program with source");
+
+        error = clBuildProgram(program, 1, &device, nullptr, nullptr, nullptr);
+        test_error(error, "Failed to build program");
+
+        kernel = clCreateKernel(program, "fill", &error);
+        test_error(error, "Failed to create copy kernel");
+
+        return CL_SUCCESS;
+    }
+
+    cl_int SetUpKernelArgs() override
+    {
+        cl_int error = CL_SUCCESS;
+        out_mem = clCreateBuffer(context, CL_MEM_WRITE_ONLY, data_size(),
+                                 nullptr, &error);
+        test_error(error, "clCreateBuffer failed");
+
+        error = clSetKernelArg(kernel, 0, sizeof(cl_int), &pattern_pri);
+        test_error(error, "clSetKernelArg failed");
+
+        error = clSetKernelArg(kernel, 1, sizeof(out_mem), &out_mem);
+        test_error(error, "clSetKernelArg failed");
+
+        return CL_SUCCESS;
+    }
+
+    cl_int SetUp(int elements) override
+    {
+        cl_int error = BasicMutableCommandBufferTest::SetUp(elements);
```

Comment:
command-buffer created here might not have `CL_COMMAND_BUFFER_SIMULTANEOUS_USE_KHR` set on creation.

Reply:
Turns out base class for mutable dispatch subset of tests ignores `simultaneous_use_requested` flag - corrected as well